src/custom/api/gnosisSafe/index.ts

```diff
@@ -19,25 +23,34 @@ const CHAIN_SHORT_NAME: Partial<Record<number, string>> = {
 
 const SAFE_TRANSACTION_SERVICE_CACHE: Partial<Record<number, SafeServiceClient | null>> = {}
 
-function _getClient(chainId: number): SafeServiceClient | null {
-  let client = SAFE_TRANSACTION_SERVICE_CACHE[chainId]
-  if (client === undefined) {
-    const url = SAFE_TRANSACTION_SERVICE_URL[chainId]
-    if (!url) {
-      client = null
-    } else {
-      client = new SafeServiceClient(url)
-    }
-
-    // Add client to cache (or null if unknonw network)
-    SAFE_TRANSACTION_SERVICE_CACHE[chainId] = client
+function _getClient(chainId: number, library: Web3Provider): SafeServiceClient | null {
+  const cachedClient = SAFE_TRANSACTION_SERVICE_CACHE[chainId]
+
+  if (cachedClient !== undefined) {
+    return cachedClient
   }
 
+  const url = SAFE_TRANSACTION_SERVICE_URL[chainId]
+  const client = url ? createSafeServiceClient(url, library) : null
+
+  // Add client to cache (or null if unknonw network)
+  SAFE_TRANSACTION_SERVICE_CACHE[chainId] = client
+
   return client
 }
 
-function _getClientOrThrow(chainId: number): SafeServiceClient {
-  const client = _getClient(chainId)
+function createSafeServiceClient(txServiceUrl: string, library: Web3Provider): SafeServiceClient {
+  const provider = new Web3Provider(library.send.bind(library) as JsonRpcFetchFunc)
```

Comment:
<img width="726" alt="image" src="https://user-images.githubusercontent.com/7122625/182582872-a08f6314-4082-4b1c-bc3a-a46a8d941b5c.png">
Because `Web3Provider` expects a specific type.
And we bind the `send` method to avoid context leak